fdbserver/DataDistribution.actor.cpp

```diff
@@ -3474,17 +3474,17 @@ ACTOR Future<Void> storageServerTracker(
 			}
 
 			if( server->lastKnownClass.machineClassFitness( ProcessClass::Storage ) > ProcessClass::UnsetFit ) {
-				// We saw a corner case in in 3 data_hall configuration
-				// when optimalTeamCount = 1, healthyTeamCount = 0.
-				if (self->optimalTeamCount > 0 && self->healthyTeamCount > 0) {
+				// NOTE: Should not use self->healthyTeamCount > 0 in if statement, which will cause status bouncing between
+				// healthy and unhealthy and result in OOM (See PR#2228).
+
+				if (self->optimalTeamCount > 0) {
 					TraceEvent(SevWarn, "UndesiredStorageServer", self->distributorId)
 					    .detail("Server", server->id)
 					    .detail("OptimalTeamCount", self->optimalTeamCount)
 					    .detail("Fitness", server->lastKnownClass.machineClassFitness(ProcessClass::Storage));
 					status.isUndesired = true;
 				}
 				otherChanges.push_back( self->zeroOptimalTeams.onChange() );
-				otherChanges.push_back(self->zeroHealthyTeams->onChange());
```

Comment:
Why was this line added before in commit a588710?

Reply:
I think @xumengpanda was trying to fix another bug indirectly and added this line. Then that bug was fixed and these two lines were not reverted. He knows more details about that bug.

Reply:
The scenario said in the removed comment is what the change was to fix, but turns out to be a workaround.
`// We saw a corner case in in 3 data_hall configuration
// when optimalTeamCount = 1, healthyTeamCount = 0.`